src/lib/ThumbnailsSidebar.js

```diff
@@ -218,30 +220,77 @@ class ThumbnailsSidebar {
             return Promise.resolve(this.thumbnailImageCache[itemIndex]);
         }
 
+        return this.getThumbnailDataURL(itemIndex + 1)
+            .then(this.createImageEl)
+            .then((imageEl) => {
+                // Cache this image element for future use
+                this.thumbnailImageCache[itemIndex] = imageEl;
+
+                return imageEl;
+            });
+    }
+
+    /**
+     * Given a page number, generates the image data URL for the image of the page
+     * @param {number} pageNum  - The page number of the document
+     * @return {string} The data URL of the page image
+     */
+    getThumbnailDataURL(pageNum) {
         const canvas = document.createElement('canvas');
 
         return this.pdfViewer.pdfDocument
-            .getPage(itemIndex + 1)
+            .getPage(pageNum)
             .then((page) => {
```

Comment:
Is it worth moving these to named class methods to avoid recreating the functions each time? It usually wouldn't matter, but I think we're doing this many times a second on scroll, right?